### ISSUE TITLE
Cannot remove member (v. 2.0.11) in CoreOS alpha (709.0.0)

### ISSUE THREAD

**Issue by eaoliver** [coreos/etcd#3009]
Trying to remove a member from a 3-node cluster, but the node won't be removed.  The node is gone. What do I have to do to get rid of it?

core@vb-command-0 ~ $ etcdctl member list
cce823088e6cee92: name=command_0 peerURLs=http://10.0.0.10:2380 clientURLs=http://localhost:2379
ece899e1a3fb5524: name=aggregator_0 peerURLs=http://10.0.231.187:2380 clientURLs=http://localhost:2379
fb92c8bbb136b542: name=rabbitmq_0 peerURLs=http://10.0.128.10:2380 clientURLs=http://localhost:2379

core@vb-command-0 ~ $ etcdctl member remove fb92c8bbb136b542
Recieved an error trying to remove member fb92c8bbb136b542: context deadline exceeded

**Comment by yichengq** [comment]
Could you set the value to the cluster, or does the cluster have quorum still?
If yes, could you share `etcdctl --debug member remove fb92c8bbb136b542` with us?

**Comment by eaoliver** [comment]
I killed two of the servers in the 3-node cluster.  Only cce823088e6cee92: name=command_0 remains.    I am trying to remove the two inactive nodes: ece899e1a3fb5524 and fb92c8bbb136b542 so that their new instances can take their place.  

I cannot set values since 2 of 3 nodes are down, which is fine.  However, I should be able to remove the inactive nodes and allow the two new nodes to join the cluster. 

From the single active node:

core@vb-command-0 ~ $ etcdctl --debug member remove fb92c8bbb136b542
Cluster-Endpoints: http://localhost:2379, http://localhost:2379, http://localhost:2379
Recieved an error trying to remove member fb92c8bbb136b542: context deadline exceeded

Here's a sample of the active node's journal output.  

Jun 19 01:05:18 vb-command-0 etcd2[571]: 2015/06/19 01:05:18 raft: cce823088e6cee92 received vote from cce823088e6cee92 at term 1193
Jun 19 01:05:18 vb-command-0 etcd2[571]: 2015/06/19 01:05:18 raft: cce823088e6cee92 [logterm: 172, index: 2273235] sent vote request to ece899e1a3fb5524 at term 1193
Jun 19 01:05:18 vb-command-0 etcd2[571]: 2015/06/19 01:05:18 raft: cce823088e6cee92 [logterm: 172, index: 2273235] sent vote request to fb92c8bbb136b542 at term 1193
Jun 19 01:05:18 vb-command-0 etcd2[571]: 2015/06/19 01:05:18 sender: dropping MsgVote because maximal number 64 of sender buffer entries to http://10.0.231.187:2380/raft has been reached
Jun 19 01:05:20 vb-command-0 etcd2[571]: 2015/06/19 01:05:20 raft: cce823088e6cee92 is starting a new election at term 1193
Jun 19 01:05:20 vb-command-0 etcd2[571]: 2015/06/19 01:05:20 raft: cce823088e6cee92 became candidate at term 1194
Jun 19 01:05:20 vb-command-0 etcd2[571]: 2015/06/19 01:05:20 raft: cce823088e6cee92 received vote from cce823088e6cee92 at term 1194
Jun 19 01:05:20 vb-command-0 etcd2[571]: 2015/06/19 01:05:20 raft: cce823088e6cee92 [logterm: 172, index: 2273235] sent vote request to ece899e1a3fb5524 at term 1194
Jun 19 01:05:20 vb-command-0 etcd2[571]: 2015/06/19 01:05:20 raft: cce823088e6cee92 [logterm: 172, index: 2273235] sent vote request to fb92c8bbb136b542 at term 1194
Jun 19 01:05:20 vb-command-0 etcd2[571]: 2015/06/19 01:05:20 sender: dropping MsgVote because maximal number 64 of sender buffer entries to http://10.0.231.187:2380/raft has been reached
Jun 19 01:05:22 vb-command-0 etcd2[571]: 2015/06/19 01:05:22 raft: cce823088e6cee92 is starting a new election at term 1194
Jun 19 01:05:22 vb-command-0 etcd2[571]: 2015/06/19 01:05:22 raft: cce823088e6cee92 became candidate at term 1195
Jun 19 01:05:22 vb-command-0 etcd2[571]: 2015/06/19 01:05:22 raft: cce823088e6cee92 received vote from cce823088e6cee92 at term 1195
Jun 19 01:05:22 vb-command-0 etcd2[571]: 2015/06/19 01:05:22 raft: cce823088e6cee92 [logterm: 172, index: 2273235] sent vote request to ece899e1a3fb5524 at term 1195
Jun 19 01:05:22 vb-command-0 etcd2[571]: 2015/06/19 01:05:22 raft: cce823088e6cee92 [logterm: 172, index: 2273235] sent vote request to fb92c8bbb136b542 at term 1195
Jun 19 01:05:22 vb-command-0 etcd2[571]: 2015/06/19 01:05:22 sender: dropping MsgVote because maximal number 64 of sender buffer entries to http://10.0.231.187:2380/raft has been reached
Jun 19 01:05:24 vb-command-0 etcd2[571]: 2015/06/19 01:05:24 raft: cce823088e6cee92 is starting a new election at term 1195
Jun 19 01:05:24 vb-command-0 etcd2[571]: 2015/06/19 01:05:24 raft: cce823088e6cee92 became candidate at term 1196
Jun 19 01:05:24 vb-command-0 etcd2[571]: 2015/06/19 01:05:24 raft: cce823088e6cee92 received vote from cce823088e6cee92 at term 1196
Jun 19 01:05:24 vb-command-0 etcd2[571]: 2015/06/19 01:05:24 raft: cce823088e6cee92 [logterm: 172, index: 2273235] sent vote request to ece899e1a3fb5524 at term 1196
Jun 19 01:05:24 vb-command-0 etcd2[571]: 2015/06/19 01:05:24 raft: cce823088e6cee92 [logterm: 172, index: 2273235] sent vote request to fb92c8bbb136b542 at term 1196
Jun 19 01:05:24 vb-command-0 etcd2[571]: 2015/06/19 01:05:24 sender: dropping MsgVote because maximal number 64 of sender buffer entries to http://10.0.231.187:2380/raft has been reached
Jun 19 01:05:25 vb-command-0 etcd2[571]: 2015/06/19 01:05:25 raft: cce823088e6cee92 is starting a new election at term 1196
Jun 19 01:05:25 vb-command-0 etcd2[571]: 2015/06/19 01:05:25 raft: cce823088e6cee92 became candidate at term 1197
Jun 19 01:05:25 vb-command-0 etcd2[571]: 2015/06/19 01:05:25 raft: cce823088e6cee92 received vote from cce823088e6cee92 at term 1197
Jun 19 01:05:25 vb-command-0 etcd2[571]: 2015/06/19 01:05:25 raft: cce823088e6cee92 [logterm: 172, index: 2273235] sent vote request to ece899e1a3fb5524 at term 1197
Jun 19 01:05:25 vb-command-0 etcd2[571]: 2015/06/19 01:05:25 raft: cce823088e6cee92 [logterm: 172, index: 2273235] sent vote request to fb92c8bbb136b542 at term 1197
Jun 19 01:05:25 vb-command-0 etcd2[571]: 2015/06/19 01:05:25 sender: dropping MsgVote because maximal number 64 of sender buffer entries to http://10.0.231.187:2380/raft has been reached

**Comment by yichengq** [comment]
The failure of removal is expected because etcd needs quorum to remove member.

The info about etcd cluster is stored in etcd store itself, so it needs to go through the write process to change the cluster configuration. This helps all etcd members see the same view of the cluster.

One way to remove it is to restart inactive one and send remove request. The other way is to go through disaster recovery: https://github.com/coreos/etcd/blob/master/Documentation/admin_guide.md#disaster-recovery

**Comment by eaoliver** [comment]
Thanks for the reply, but I don't think you guys have thought about how users actually use etcd2.  

I would add the ability to force-new-cluster straight from etcdctl without having to stop and start the etcd2 service, which I suspect is how most people run etcd2 on CoreOS.  

I should be able to do etcdctl --force-new-cluster on one machine, and then do something like etcdctl --reset on the other nodes.  --reset would flush the current state, reconnect to the discovery service, and re-join the cluster.  

The name collisions with the discovery service also need to go away.  Why do names have to be unique when the nodes have their own random ID?  I stopped a service and relaunched a new one and am getting entries like this in the journal:

Jun 19 04:30:35 rabbitmq_0 etcd2[572]: 2015/06/19 04:30:35 member "rabbitmq_0" has previously registered with discovery service token (https://discovery.etcd.io/e7e6e52865eb5e55fc06139c4d76204d).

**Comment by xiang90** [comment]
> Thanks for the reply, but I don't think you guys have thought about how users actually use etcd2.

I do not think you really understand how reconfiguration works and why it works as it is today. 
Runtime reconfiguration is not a simple task and it introduces tons of hard problems you might not think of. I think we made the best trade-off between flexibility and safety. We set the line very clear that etcd do not support any unsafe operation at runtime; each reconfiguration must be set by user explicitly. Our users are happy about today's reconfiguration setup and we have no plan to change it.

> The name collisions with the discovery service also need to go away.

The discovery service SHOULD be only used to bootstrap your cluster. It will not reflect the runtime configuration of your cluster.
